client/src/pages/LyricsView/LyricsDisplay.tsx

```diff
@@ -52,46 +79,126 @@ const LyricsDisplay: React.FunctionComponent<IProps> = ({
     }
   }, [searchShown]);
 
+  //accounting for rewinding the song
+  useEffect(() => {
+    if (progressMs === undefined) {
+      return; // song switched
+    }
+    if (paused) {
+      return; // song is paused
+    }
+    if (progressMs < naturalSongProgress) {
+      const clonedLyricsArray = syncedLyricsArray === undefined ? [] : [...syncedLyricsArray];
+      setNaturalSongProgress(0);
+      setSyncedLyrics({ array: clonedLyricsArray, before: "", highlighted: "", after: lyrics });
+    }
+  }, [progressMs]);
+
+  //making sure highlighted text is always visible
+  useEffect(() => {
+    const element = highlightedRef.current;
+    if (element) {
+      element.scrollIntoView({ behavior: "smooth", block: "center", inline: "nearest" });
+    }
+  }, [syncedLyricsArray, progressMs]);
```

Comment:
I think it might make more sense if this useEffect just triggered on `syncedLyrics` change as we want to move the screen when the displayed lyrics change.

```suggestion
  }, [syncedLyrics]);
```